addons/mrp_account/models/product.py

```diff
@@ -44,17 +45,24 @@ def _compute_average_price(self, qty_invoiced, qty_to_invoice, stock_moves):
         self.ensure_one()
         if stock_moves.product_id == self:
             return super()._compute_average_price(qty_invoiced, qty_to_invoice, stock_moves)
-        bom = self.env['mrp.bom']._bom_find(product=self, company_id=stock_moves.company_id.id, bom_type='phantom')
+        if stock_moves:
+            product_moves = defaultdict(lambda: self.env['stock.move'])
+            for sm in stock_moves:
+                product_moves[sm.product_id] |= sm
+            value = 0
+            for product, moves in product_moves.items():
+                qty = sum(moves.mapped('product_qty'))
```

Comment:
The test condition above doesn't multiply qty_invoiced with stock_move.product_qty 
Why do you need to do it here ?